addon/addon.go

```diff
@@ -166,3 +167,29 @@ func NewAddonManager(kubeConfig *rest.Config) (addonmanager.AddonManager, error)
 	err = addonMgr.AddAgent(agentAddon)
 	return addonMgr, err
 }
+
+func startAddon(ctx context.Context) {
+	kubeConfig, err := ctrl.GetConfig()
+	if err != nil {
+		klog.Error(err, "unable to get kubeConfig , addon cannot be installed", "controller", "SearchOperator")
+		return
+	}
+	addonMgr, err := NewAddonManager(kubeConfig)
+	if err != nil {
+		klog.Error(err, "unable to create a new  addon manager", "controller", "SearchOperator")
+	} else {
+		klog.Info("starting search addon manager")
+		err = addonMgr.Start(ctx)
+		if err != nil {
+			klog.Error(err, "unable to start a new  addon manager", "controller", "SearchOperator")
+		}
+	}
+}
+
+func CreateAddonOnce(ctx context.Context, instance *searchv1alpha1.Search) {
```

Comment:
Can you add a quick description for this function?  I'm looking for why "Once" ?

Reply:
Done